### PR TITLE
Rebalance CI jobs

### PR DESCRIPTION
**Pull request by andrewpbrett**
#### What? Why?

Related to #6752

Rebalances the CI jobs so that the build finishes faster.

This also pushes us to 7 jobs per build, with 20 workers. One reliably finishes in about 5 minutes, so even in the event where 3 builds start exactly simultaneously, they would still all finish in about 12 minutes, since a worker from the first build is freed up in time to start on the third build. I've also never seen any build get stuck waiting on another one to finish so far. 

#### What should we test?

Green build, maybe in about 12-15 minutes? 

#### Release notes

Rebalanced CI jobs to speed up test suite. 

Changelog Category: Technical changes

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
There's a weird looking failure on an API controller test there...

Otherwise, it looks awesome :tada:

**Comment by Matt-Yorkley** [comment]
Hmm it's failing consistently, and only on this PR:
```
1) Api::Admin::OrderCycleSerializer serializes the order cycle with editable_variants_for_incoming_exchanges
     Failure/Error: expect(serializer.editable_variants_for_incoming_exchanges.to_json).to_not include order_cycle.distributors.first.id.to_s
       expected "{\"144\":[146,147],\"145\":[148,149]}" not to include "146"
     # ./spec/serializers/api/admin/order_cycle_serializer_spec.rb:21:in `block (2 levels) in <module:Admin>'
```

not sure why :face_with_head_bandage:

**Review by Matt-Yorkley — APPROVED** [review]

**Comment by andrewpbrett** [comment]
I made a mistake when stuffing shuffling the tests around, so that serializer spec was actually running twice... not sure why it would matter, but fixing the mistake got it back to green. I'm shifting one last thing around in this latest push and then we'll hopefully be at about 13 minutes. There's a variant override spec that still looks a little flaky, I've seen it elsewhere though.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -90,9 +131,9 @@ jobs:
           bundle exec rake db:schema:load RAILS_ENV=test
 
       - name: Run admin feature tests
-        run: bundle exec rspec --profile -- spec/features/admin/*_spec.rb
+        run: bundle exec rspec --profile -- spec/features/admin/[a-o]*_spec.rb
```
Comment (mkllnk) [inline .github/workflows/build.yml:134]: Should we include `0-9` in case someone adds a `0_downtime_spec`? :grin:

**Review by mkllnk — APPROVED** [review]
Great, let's do this.

I'm a bit concerned about the overall approach though. In another pull request, @Matt-Yorkley added a file under `spec/migrations` and this config would miss that, right? I think that we will need a way to ensure that all specs are included. Automatic rebalancing would be nice, too.

**Pull request merged**